x/bep3/types/params.go

```diff
@@ -0,0 +1,129 @@
+package types
+
+import (
+	"fmt"
+
+	"github.com/cosmos/cosmos-sdk/x/params"
+)
+
+// Parameter keys
+var (
+	KeyMinLockTime     = []byte("MinLockTime")
+	KeyMaxLockTime     = []byte("MaxLockTime")
+	KeySupportedAssets = []byte("SupportedAssets")
+
+	AbsoluteMaximumLockTime int64 = 2592000000000000 // 30 days
+	DefaultMinLockTime      int64 = 43200000000000   // 12 hours
+	DefaultMaxLockTime      int64 = 604800000000000  // 7 days
+	DefaultSupportedAssets        = AssetParams{AssetParam{Denom: "kava", CoinID: "459", Limit: 1, Active: false}}
+)
+
+// Params governance parameters for bep3 module
+type Params struct {
+	MinLockTime     int64       `json:"min_lock_time" yaml:"min_lock_time"`       // HTLT minimum lock time
+	MaxLockTime     int64       `json:"max_lock_time" yaml:"max_lock_time"`       // HTLT maximum lock time
+	SupportedAssets AssetParams `json:"supported_assets" yaml:"supported_assets"` // supported assets
+}
+
+// String implements fmt.Stringer
+func (p Params) String() string {
+	return fmt.Sprintf(`Params:
+	Min lock time: %d,
+	Max lock time: %d,
+	Supported assets: %s`,
+		p.MinLockTime, p.MaxLockTime, p.SupportedAssets)
+}
+
+// NewParams returns a new params object
+func NewParams(minLockTime int64, maxLockTime int64, supportedAssets AssetParams) Params {
+	return Params{
+		MinLockTime:     minLockTime,
+		MaxLockTime:     maxLockTime,
+		SupportedAssets: supportedAssets,
+	}
+}
+
+// DefaultParams returns default params for bep3 module
+func DefaultParams() Params {
+	return NewParams(DefaultMinLockTime, DefaultMaxLockTime, DefaultSupportedAssets)
+}
+
+// AssetParam governance parameters for each asset within a supported chain
+type AssetParam struct {
+	Denom  string `json:"denom" yaml:"denom"`     // name of the asster
+	CoinID string `json:"coin_id" yaml:"coin_id"` // internationally recognized coin ID
```

Comment:
Bip44 coin type often doesn't differentiate between testnet and mainnets, so we might need another way to identify different chains. Cosmos chains all have a unique "chain-id", but I'm not sure how to create identifiers for things like eth and eth testnets.

Reply:
Was hoping that Cosmos would follow Ethereum's standard of renaming coin Denom to indicate the network, e.g. Eth (mainnet), rEth (ropsten), and kEth (kovan), but looks like that is not the case with Cosmos.

Reply:
Previously the param hierarchy looked like []chain -> []assets, such that each chain had a unique ID and this was not a problem. The params have been flattened such that the bep3 module only supports one chain (i.e. the binance testnet) and assets exist under that chain. I'm very open to changing the params, but the current param structure is much simpler and also still enables us to launch the 'cross-chain testnet' from bnbchain testnet <-> kava testnet.